### PR TITLE
add indentation options to kate

### PR DESCRIPTION
**Pull request by Asqiir**
This is the first contribution to #87 .
It allows users to configure the way kate handles indentation.

# Reasoning about decisions:
I  decided to put all options into `config.programs.kate.editor`. The `editor` option should contain all options that act on configuration into the rectangle where users type text in (as opposed to: the console, the `project` tab, the menus and sidebards).

# Known limitations:
* `kate` needs to be restart after switching the configuration. I could not figure out how to make `kate` do a hot reload on the changed `katerc`.
* I would like to have the message in the assertion reference the variables in question not hard-coded by their name. I would prefer if the compilation fails if someone was to change the names of the variable but forgot to update the message.

Do you want me to put all changes into a single MR (leaving this one open), or do you prefer each change to be its own PR?

### CODE REVIEW

**Comment by Asqiir** [comment]
Should we add the `katerc` (`.config/katerc`) into the `defaultResetFiles` in [`modules/files.nix`](https://github.com/pjones/plasma-manager/blob/trunk/modules/files.nix)? (I'm not sure what that list does.)

File: modules/apps/kate.nix [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  # compute kate's magic TabHandlingMode
+  boolToInt = b: if b then 1 else 0;
+  t = boolToInt config.programs.kate.editor.indent.tabFromEverywhere;
+  s = boolToInt config.programs.kate.editor.indent.undoByShiftTab;
+  tabHandlingMode = t + 2 * s - 1;
+in
+{
+  options.programs.kate.editor = {
+    tabWidth = lib.mkOption {
+      description = "The width of a single tab (\\t) sign (in number of spaces).";
```
Comment (magnouvean) [inline modules/apps/kate.nix]: Shouldn't it just be \t here and not \\\t?

File: modules/apps/kate.nix [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      default = true;
+      type = lib.types.bool;
+    };
+  };
+
+  config.assertions = [
+    {
+      assertion = tabHandlingMode >= -1;
+      message = "Either 'undeByShiftTab' or 'tabFromEverywhere' needs to be enabled!";
+      # kate does not seem to support disabling both.
```
Comment (magnouvean) [inline modules/apps/kate.nix]: I would put the comment above the message. Maybe move it up and rewrite as something like "kate doesn't seem to support disabling undeByShiftTab and tabFromEverywhere at the same time"

**Review by magnouvean — COMMENTED** [review]
See my two small comments. Looks good to me otherwise assuming you have tested that the module works (I don't really use kate so I can't really test)

File: modules/apps/kate.nix (hunk 2 above)
Comment (Asqiir) [reply modules/apps/kate.nix]: I changed the error message.

File: modules/apps/kate.nix (hunk 1 above)
Comment (Asqiir) [reply modules/apps/kate.nix]: `\t` is rendered to the tab whitespace.
`\''t` is rendered to the tab character. I switched to that.

Source: [the nix documentation](https://nixos.org/manual/nix/stable/language/values.html#type-string)

**Comment by Asqiir** [comment]
~~found bugs :( 
(do not merge yet, I hope to fix it today)~~

fixed

**Comment by magnouvean** [comment]
I added an enable option (which is in general good practice) and assigned `config.programs.kate` to a variable `cfg` (which we do throughout the project to shorten things up a bit). See my two earlier commits. Are you happy enough with this so that we can merge this now @Asqiir?

**Comment by Asqiir** [comment]
@magnouvean we can merge this imo

**Pull request merged**